src/common/CommonMonthSolve.tsx

```diff
@@ -1,32 +1,63 @@
+import { useEffect, useState } from 'react';
 import styled from 'styled-components';
 import CommonCalendar from '../components/Calendar/Calendar';
+import useGetMonthSolve from '../libs/hooks/Home/useGetMonthSolve';
 
 const CommonMonthSolve = () => {
   const currentMonth = new Date().getMonth() + 1; // getMonth()는 0부터 시작하므로 +1 필요
   const monthText = `${currentMonth}월 문제풀이 현황`;
-  const TotalSolve = 13;
-  const LongestSolveDay = 3;
-  const LongestSolvedCount = 2;
+
+  const [stats, setStats] = useState({
+    totalCount: 0,
+    longestPeriod: 0,
+    maxCount: 0,
+  });
+
+  const year = new Date().getFullYear();
+  const [clickedYear, setClickedYear] = useState(year);
+  const [clickedMonth, setClickedMonth] = useState(currentMonth);
+
+  const { data } = useGetMonthSolve({
+    year: clickedYear,
+    month: clickedMonth,
+  });
+
+  useEffect(() => {
+    if (data) {
+      const { totalCount, longestPeriod, maxCount } = data.data;
+      setStats({
+        totalCount,
+        longestPeriod,
+        maxCount,
+      });
+    }
+  }, [data]);
```

Comment:
P2: 아래처럼 구현하면 불필요한 useState + useEffect를 정의하지 않아도 됩니다 !

```suggestion
  const { data, isLoading } = useGetMonthSolve({
    year: clickedYear,
    month: clickedMonth,
  });
  const { totalCount, longestPeriod, maxCount } = !isLoading && data?.data;
```

Reply:
오 원리가 좀 신기하네요.. 찾아보겠습니다 감사합니다